### PR TITLE
Refinements for importing transition lists

### PR DESCRIPTION
**Pull request by henrytsanford**
Changes to how Skyline imports transition lists. Me and Henry Estberg's recent changes include:

Adding automatic detection for Label Type, Fragment Name, and Precursor Charge columns
User selected headers from the last document are now reused when the user pastes in a CSV file. These headers are only reused when the column count and headers, provided there are any, match those of the previous file.
Changing workflow and button layout of the column selection screen.
Expanding on PasteTransitionListTest in order to cover our additional features and better test preexisting features.

### CODE REVIEW

File: pwiz_tools/Skyline/TestUtil/TestFunctional.cs [hunk 1]
```diff
@@ -1,2240 +1,2283 @@
-﻿/*
```
Comment (bspratt) [inline pwiz_tools/Skyline/TestUtil/TestFunctional.cs:1]: This one looks like just a line endings change. I'll see if I can clean that up.

Comment (bspratt) [reply pwiz_tools/Skyline/TestUtil/TestFunctional.cs:1]: ...but of course when I compare it to master, WinMerge doesn't show any line ending or other whitespace change, There are a handful of actual changes, so I'll try to sort this out.

**Comment by nickshulman** [comment]
When I import a transition list that I wrote, all I get is a message that "Failed to find molecule column".
I was expecting that the dialog would come up and help me match the columns up to what is in the text file.
Am I misusing this feature? Is there a minimum level of conformity that the transition list needs to have in order to be able to get to the ImportTransitionListColumnSelectDlg?

[testtransitionlist.txt](https://github.com/ProteoWizard/pwiz/files/5392987/testtransitionlist.txt)

**Comment by bspratt** [comment]
Skyline recognizes that as a broken small molecule transition list. The present work is for dealing with proteomic transition lists. Ideally we will extend this enhanced column picking interface to small molecules as well.

**Comment by nickshulman** [comment]
Do you want to make the grid with the combo boxes that is on the ImportTransitionListColumnSelectDlg into a UserControl that can be used in other places? You might find that doing that makes it easier to separate all of the logic for displaying that sort of grid from the transition list-specific stuff.

**Comment by bspratt** [comment]
That's an excellent plan for moving forward to a more general use of this work. For now, though, Henry and Henry are back in school and just want to land this thing. But that should probably go in as a // CONSIDER comment.

**Comment by bspratt** [comment]
Some post-merge test failures here that need investigating. I'm going to make this a Draft PR for now.

File: pwiz_tools/Skyline/FileUI/ImportTransitionListErrorDlg.cs [hunk 2]
```diff
@@ -45,17 +50,48 @@ public ImportTransitionListErrorDlg(List<TransitionImportErrorInfo> errorList, b
             // If all of the transitions were errors, canceling and accepting are the same
             // so give a different message and disable the cancel button
             string errorListMessage;
-            if (isErrorAll)
+            // If the transition list is missing essential columns, tell the user in a 
+            // readable way
+            if (MissingEssentialColumns.Count != 0)
+            {
+                errorListMessage = @"Your transition list must contain ";
```
Comment (bspratt) [inline pwiz_tools/Skyline/FileUI/ImportTransitionListErrorDlg.cs:57]: These strings should be localized. I'll fix that given that time is short.

File: pwiz_tools/Skyline/SkylineFiles.cs [hunk 3]
```diff
@@ -1855,11 +1855,27 @@ public void ImportMassList(MassListInputs inputs, string description, bool assay
...
+                return;
+
+            using (var columnDlg = new ImportTransitionListColumnSelectDlg(importer, docCurrent, inputs, insertPath))
+            {
+                var result = columnDlg.ShowDialog(this);
+                if (result == DialogResult.Cancel)
+                    return;
+            }
+            
+            // User feedback
```
Comment (bspratt) [inline pwiz_tools/Skyline/SkylineFiles.cs:1871]: Do we know what these comments are about?

File: pwiz_tools/Skyline/SkylineFiles.cs [hunk 4]
```diff
@@ -1855,11 +1855,27 @@ public void ImportMassList(MassListInputs inputs, string description, bool assay
             List<PeptideGroupDocNode> peptideGroups = null;
             var docCurrent = DocumentUI;
             SrmDocument docNew = null;
+
+            // Play around here with PreImport of mass list
```
Comment (bspratt) [inline pwiz_tools/Skyline/SkylineFiles.cs:1859]: Drop the "play around", I think

File: pwiz/utility/misc/Stream.hpp [hunk 5]
```diff
@@ -57,7 +57,9 @@ using std::streampos;
 using std::streamoff;
 using std::streamsize;
 
+/* this breaks VS2019 usage
```
Comment (bspratt) [inline pwiz/utility/misc/Stream.hpp:60]: We should revert this change before merging to master.

Comment (HenryEstberg) [reply pwiz/utility/misc/Stream.hpp]: The line of code is now uncommented out, I left the comment on breaking VS2019 in case it comes up again.

File: pwiz_tools/Skyline/SkylineFiles.cs (hunk 4 above)
Comment (HenryEstberg) [reply pwiz_tools/Skyline/SkylineFiles.cs]: Done

**Review by bspratt — APPROVED** [review]
OK, looks good. We've been over it together several times in Zoom.
